### PR TITLE
fix(wizard): reworked wizard so it can be used in modal

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/2046

@tlabaj @mcarrano this is pretty straight forward except for one thing. The wizard when used in a modal needs a defined height, otherwise the height will be defined by the wizard contents and will change as a user progresses through the wizard. There are a couple of ways we can do this.

1) Assign a height to the wizard for use when it's in a modal, then modify that height when the wizard is used in context of the page (the "in page" variation). This seems to work just fine and allows us to define a specific height for the wizard (that the user can override), but it means that we need to maintain separate variations of the wizard for use in the page and in a modal. The user impact is that the user has to specify the modal width when using a modal, and they have to specify the "in page" variation when using the modal in a page.

2) Add an enhancement to the modal to define a modal height (that is overridable by the user, similar to the height variations we currently offer) and allow that to set the wizard modal height. The wizard can then simply occupy the height of the container it's in both the modal and in page variations, meaning we could do away with the in page wizard variation. The user impact of this is that the user must specify both the width and height when using a modal, but doesn't have to specify a variation of the wizard to use in either context.

WDYT?

### CODE REVIEW

**Comment by mcarrano** [comment]
@mcoker I'm having a bit of a hard time understanding the examples and what I should be looking for. As to your question, IMO the second option seems like a cleaner approach. 

Just another small thing, a while ago we decided that the Cancel button should be spaced further away from the Next and Back buttons. I don't see that here. Here is the most recent visual spec:

Wizard spec

**Comment by mcoker** [comment]
@mcarrano the cancel button spacing was put on hold in favor of the action list component - https://github.com/patternfly/patternfly/issues/2021.

**Comment by mcarrano** [comment]
Yes, I do recall that @mcoker . Do you know if this is on the roadmap? It was defined a while ago and I am surprised it was never implemented.

**Comment by mcoker** [comment]
@mcarrano it's slated for the 2020.08 milestone - June 23.

**Comment by mcoker** [comment]
@mcarrano this PR is ready for review. Our modal now behaves like a card, where if you simply omit the title/body/footer, it’s a box (with no padding) to put things in. Users only have to specify the size variation of the modal to use (large). The wizard detects that it is in a modal and retains its existing height (762px), and shrinks with the window. I tested on large and small screens and it looks good. There is only 16px padding around the window when the modal shrinks to fit the window size, so not much real estate lost from how the existing wizard goes to the edges of the window on small screens.

File: src/patternfly/components/Wizard/wizard.scss [hunk 1]
```diff
@@ -1,23 +1,6 @@
...
-  --pf-c-wizard--m-in-page--Height: 100%;
-  --pf-c-wizard--m-in-page--Width: auto;
-  --pf-c-wizard--m-in-page--lg--MaxWidth: none;
-  --pf-c-wizard--m-in-page--lg--MaxHeight: none;
-
-  @media screen and (min-width: $pf-global--breakpoint--lg) {
-    --pf-c-wizard--Width: var(--pf-c-wizard--lg--Width);
-    --pf-c-wizard--Height: var(--pf-c-wizard--lg--Height);
-  }
+  --pf-c-modal-box--c-wizard--Height: #{pf-size-prem(762px)};
```
Comment (christiemolloy) [inline src/patternfly/components/Wizard/wizard.scss:3]: this is such a specific height, maybe add a comment explaining why it is this height

**Comment by christiemolloy** [comment]
Release notes look great!

**Pull request merged**